File: 0924-fair-candy-swap/0924-fair-candy-swap.cpp
class Solution {
public:
    vector<int> fairCandySwap(vector<int>& aliceSizes, vector<int>& bobSizes) {
        int sumA = accumulate(aliceSizes.begin(), aliceSizes.end(), 0);
        int sumB = accumulate(bobSizes.begin(), bobSizes.end(), 0);

        int diff = (sumA - sumB)/2;

        unordered_set<int> setB(bobSizes.begin(), bobSizes.end());

        for (auto x : aliceSizes){
            int y = x - diff;

            if (setB.count(y)){
                return {x,y};
            }
            
        }
        return {};
        
    }
};

File: 0924-fair-candy-swap/0924-fair-candy-swap.py
class Solution:
    def fairCandySwap(self, aliceSizes: List[int], bobSizes: List[int]) -> List[int]:
        sumA = sum(aliceSizes)
        sumB = sum(bobSizes)

        diff = (sumA - sumB)//2
        l=[]

        setB = set(bobSizes)

        for x in set(aliceSizes):
            y= x - diff
            if y in setB:
                return [x,y]